### DISCUSSION TITLE
Force which GPU to use

### DISCUSSION THREAD

**Discussion by ManuelaPa** [tox-dev/tox#2258] [Q&A] [answered]
Hello,

I'm using tox to create a virtual env, but the unittest are executed using the default GPU and I want to use the other GPU. I tried to set the second GPU using : setenv = CUDA_VISIBLE_DEVICES="1" but the tests are executed in the other GPU. How can I configure my tox to be able to choice the GPU?

Thank you

**Comment by jugmac00** [comment, marked as answer]
I think this is nothing which `tox` can influence. Which (cli)app do you run with `tox`?

**Reply by ManuelaPa** [reply]
hello, i'm using the command line with two commands: 
activate envpython
tox
And then in my tox I'm launching some unittest that doesn't use the right gpu, when I configure my env installing by hand the packages the tests passes without any trouble

**Reply by jugmac00** [reply]
I googled a bit and found a couple of `tox.ini` files, and apart from also specifying `CUDA_VISIBLE_DEVICES` I see nothing special.

I am afraid I cannot help you.

Maybe it would help if you post your complete `tox.ini` file so others can help you?

**Comment by gaborbernat** [comment]
I feel this question is not tox specific so I'll lock it.
